File: CodeChef/LongContest/2013/August/hello.cpp
#include <bits/stdc++.h>

using namespace std;

inline
int compareTo(double x, double y, double tol=1e-9) {
  return (x <= y + tol) ? (x + tol < y) ? -1 : 0 : 1;
}

long long lcm(long long a, long long b) {
  return (a * b) / __gcd(a,b);
}

int n;
long long b, m[111],a[111];
double v, p[111];

int main() {
  int t;
  scanf("%d", &t);

  while (t--) {
    int x, y;
    scanf("%lf %lld %d", &v, &b, &n);

    long long total_months = 1;
    for (int i = 0; i < n; i++) {
      scanf("%lld %lf %lld", m+i, p+i, a+i);
      total_months = lcm(total_months, m[i]);
    }

    double best = total_months * v * b;
    int ans = 0;

    for (int i = 0; i < n; i++) {
      long long periods = total_months / m[i];
      double curr = periods * (m[i] * p[i] * b + a[i]);

      if (compareTo(curr, best) < 0) {
        ans = i+1;
        best = curr;
      }
    }

    printf("%d\n", ans);
  }

  return 0;
}

File: CodeChef/LongContest/2013/August/hello.py
from fractions import *
from sys import stdin

def lcm(a,b):
  return (a * b) / gcd(a, b)

lines = stdin.readlines()
t = int(lines[0])
pos = 1
for tc in xrange(t):
  v, b, n = map(int, ''.join(lines[pos].split('.')).split())
  pos += 1

  m, p, a = [0]*n, [0]*n, [0]*n
  total = 1
  for i in xrange(n):
    m[i], p[i], a[i] = map(int, ''.join(lines[pos].split('.')).split())
    a[i] = a[i] * 100
    total = lcm(total, m[i])
    pos += 1

  best = total * v * b
  ans = 0

  for i in xrange(n):
    period = total / m[i]
    curr = period * (m[i] * p[i] * b + a[i])
    if curr < best:
      ans = i+1
      best = curr

  print ans
